src/alembic/versions/75857bf0c826_introduce_officer_tables.py

```diff
@@ -0,0 +1,108 @@
+"""introduce officer tables
+
+Revision ID: 75857bf0c826
+Revises: 0db2c57ce969
+Create Date: 2024-06-03 06:36:40.642476
+
+"""
+
+from collections.abc import Sequence
+from typing import Union
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision: str = "75857bf0c826"
+down_revision: str | None = "0db2c57ce969"
+branch_labels: str | Sequence[str] | None = None
+depends_on: str | Sequence[str] | None = None
+
+
+def upgrade() -> None:
+    # drop all existing user session data
+    # TODO: combine all past migrations into this one
+    op.drop_table("user_session")
+    op.create_table(
+        "user_session",
+        sa.Column("session_id", sa.String(512), nullable=False, primary_key=True),
+        sa.Column("issue_time", sa.DateTime, nullable=False),
+        sa.Column("computing_id", sa.String(32), nullable=False),
+    )
+    op.create_unique_constraint("unique__user_session__session_id", "user_session", ["session_id"])
+    op.create_unique_constraint("unique__user_session__computing_id", "user_session", ["computing_id"])
+
+    # drop all existing site user data
+    # TODO: combine all past migrations into this one
+    op.drop_table("site_user")
+    op.create_table(
+        "site_user",
+        sa.Column("computing_id", sa.String(32), nullable=False, primary_key=True),
+    )
+    op.create_unique_constraint("unique__site_user__computing_id", "site_user", ["computing_id"])
+    op.create_foreign_key(
+        "fk__site_user__user_session__computing_id", "site_user", "user_session", ["computing_id"], ["computing_id"]
```

Comment:
Is this a foreign key from site_user -> user_session? I think it should be the other way around, because once a session is stale, won't it be deleted from user_session? Thereby making invalid all occurrences of that computing ID in site_user?

Reply:
https://app.diagrams.net/#G11Tqg_PGDp6AeEZZTMg8Ebt5_vCzZUfSz#%7B%22pageId%22%3A%22uIugNh01Jqb4SScGoUdQ%22%7D

sqlachemy is a bit funky, but it's actually the opposite. This is a foreign key stored in site_user, that will be accessed by user_session.